configuration/logger.go

```diff
@@ -1,60 +1,15 @@
 package configuration
 
 import (
-	"io"
-	"log"
+	"github.com/sirupsen/logrus"
 	"os"
 )
 
-type Logger struct {
-	debug  *log.Logger
-	info   *log.Logger
-	warn   *log.Logger
-	error  *log.Logger
-	writer io.Writer
-}
-
-func NewLogger(prefix string) *Logger {
-	writer := io.Writer(os.Stdout)
-	logger := log.New(writer, prefix, log.Ldate|log.Ltime|log.Lshortfile)
-
-	return &Logger{
-		debug:  log.New(writer, "DEBUG: ", logger.Flags()),
-		info:   log.New(writer, "INFO: ", logger.Flags()),
-		warn:   log.New(writer, "WARN: ", logger.Flags()),
-		error:  log.New(writer, "ERROR: ", logger.Flags()),
-		writer: writer,
-	}
-}
-
-func (l *Logger) Debug(value ...interface{}) {
-	l.debug.Println(value...)
-}
-
-func (l *Logger) Info(value ...interface{}) {
-	l.info.Println(value...)
-}
-
-func (l *Logger) Warn(value ...interface{}) {
-	l.warn.Println(value...)
-}
-
-func (l *Logger) Error(value ...interface{}) {
-	l.error.Println(value...)
-}
-
-func (l *Logger) Debugf(format string, value ...interface{}) {
-	l.debug.Printf(format, value...)
-}
-
-func (l *Logger) Infof(format string, value ...interface{}) {
-	l.info.Printf(format, value...)
-}
-
-func (l *Logger) Warnf(format string, value ...interface{}) {
-	l.warn.Printf(format, value...)
-}
+func NewLogger() *logrus.Logger {
+	logger := logrus.New()
+	logger.SetFormatter(&logrus.JSONFormatter{})
+	logger.SetOutput(os.Stdout)
+	logger.SetReportCaller(true)
```

Comment:
recomendation: I don't see the point of creating the logger every time. You can configure your logs in your main once and then use it throughout your code without calling `New` every time.

When you use `SetFormatter` on your global logger, it is set to the logrus logger and becomes accessible over all your application.

https://github.com/sirupsen/logrus/blob/master/logger.go#L384

See ex here: https://pkg.go.dev/github.com/sirupsen/logrus#section-readme